core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/AwsS3ProtocolFactory.java

```diff
@@ -41,6 +51,12 @@ private AwsS3ProtocolFactory(Builder builder) {
         return Optional.of(document);
     }
 
+    private Optional<XmlElement> getErrorRootFromSuccessBody(XmlElement document) {
+        return ERROR_IN_SUCCESS_BODY_ELEMENT_NAME.equals(document.elementName())
+            ? Optional.of(document)
```

Comment:
Does this follow our checkstyle?

Reply:
Yes...? Are you concerned that I'm calling a method (equals) on a String constant? I consider that good practice because it mitigates the NPE that could be thrown if document.elementName() happens to be null.

Reply:
Nope, was on the location of the ? on the new line. Thought we were more indented usually

Reply:
It did pass checkstyle but I changed it anyway, see if you like the new version more.